testing/src/main/java/io/grpc/FrozenClock.java

```diff
@@ -180,28 +193,43 @@ public String toString() {
   }
 
   /**
-   * Provides a partially implemented instance of {@link ScheduledExecutorService} that uses the
-   * fake clock ticker for testing.
+   * Returns a partially implemented instance of {@link ScheduledExecutorService} that uses this
+   * clock ticker for testing.
+   *
+   * <p>Only {@link ScheduledExecutorService#schedule(Runnable, long, TimeUnit)} and
+   * {@link ScheduledExecutorService#execute(Runnable)} are supported.
    */
   public ScheduledExecutorService getScheduledExecutorService() {
     return scheduledExecutorService;
   }
 
   /**
-   * Provides a stopwatch instance that uses the fake clock ticker.
+   * Provides a stopwatch instance that uses the clock ticker.
    */
   public Supplier<Stopwatch> getStopwatchSupplier() {
     return stopwatchSupplier;
   }
 
   /**
-   * Ticker of the FakeClock.
+   * Ticker of the clock.
    */
   public Ticker getTicker() {
     return ticker;
   }
 
   /**
+   * Returns a {@link Deadline} whose remaining time is controlled by this class.
+   *
+   * <p>If the deadline is to be passed into
+   * {@link Context#withDeadline(Deadline, ScheduledExecutorService)}, then
+   * {@link #getScheduledExecutorService()} should be used as the executor to ensure deadline
+   * cancellations are not scheduled based on system time.
+   */
+  public Deadline createDeadlineAfter(int duration, TimeUnit unit) {
+    return Deadline.after(duration, unit, deadlineTicker);
```

Comment:
This is not safe, as mixing "normal" Deadlines and "fake" Deadlines will lead to undefined/weird results. This is why Ticker wasn't exposed earlier.